### PR TITLE
Added LazyDataLoader Functionality

### LINKED ISSUE #182: `LazyDataLoader` that loads batches of data lazily
## Feature Description
Currently, we have `DataLoader` abstract like class (along with other concrete implementations in `TensorDataLoader` and `MultiTensorDataLoader`  in SwiftSyft that is analogous to PyTorch's `torch.utils.data.DataLoader`. These classes take in a collection of tensors, shuffles them and batches them together. The `SwiftSyft` implementation of DataLoaders are equal to a PyTorch `DataLoader` that takes a map-style dataset since those dataset types can be shuffled.

Another set of `DataLoader`s we need will be ones that can take in a lazily evaluated sequence of elements. Similar to the current `DataLoader`, the new `LazyDataLoader` will instead take in a Swift `Sequence` of `TorchTensor`s that cannot be shuffled (because they are lazily evaluated) and batch them similar to the logic in the current implementation of SwiftSyft `DataLoader`.

## API

- [ ] Abstract class `LazyDataLoader` (similar to current `DataLoader` but takes in a `Sequence`)
- [ ] Concrete subclass `LazyTensorDataLoader` (takes a `Sequence` of tensors and batches them. There will be no `shuffle` parameter)
- [ ] Concrete subclass `LazyMultiTensorDataLoader` (takes a `Sequence` of arrays of tensors and batches them. There will be no`shuffle` parameter)

## What alternatives have you considered?
Considered implementing all the logic in one class but it was too difficult by just using generics.

### PR DESCRIPTION
**Pull request by Rohith-hacker**
## Description
- Added LazyDataLoader class which takes in a lazily evaluated sequence of elements. 
- Added Test file for LazyDataLoader Class
- refactored CycleRequestTests file 
closes #182 

## Affected Dependencies
None

## How has this been tested?
LazyDataLoaderTests added. (more tests to be added after libtorch support for cpu devices)

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [X] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Review by mjjimenez — CHANGES_REQUESTED** [review]
Looks good. One thing though is that I don't think we need `LazyRandomIterator` anymore. It doesn't become lazy anymore since we store the sequence in an array and we evaluate the entire sequence and store it into memory. We can just use the default `Sequence` Iterator like you did in the `LazyDataLoaders`

**Comment by Rohith-hacker** [comment]
@mjjimenez yes, removed `LazyRandomIterator` and used default `Sequence` iterator

**Review by mjjimenez — APPROVED** [review]
